File: qqbot-module-qqbot-runtime/src/main/java/com/mieai/qqbot/app/outbox/OutboxRuntimeProperties.java
package com.mieai.qqbot.app.outbox;

import java.time.Duration;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties("qqbot.outbox")
public class OutboxRuntimeProperties {
    private boolean enabled = true;
    private Duration pollInterval = Duration.ofSeconds(1);
    private Duration leaseDuration = Duration.ofSeconds(45);
    private Duration requestTimeout = Duration.ofSeconds(20);
    private int maxAttempts = 8;
    private int batchSize = 16;

    public boolean isEnabled() { return enabled; }
    public void setEnabled(boolean enabled) { this.enabled = enabled; }
    public Duration getPollInterval() { return pollInterval; }
    public void setPollInterval(Duration value) { pollInterval = value; }
    public Duration getLeaseDuration() { return leaseDuration; }
    public void setLeaseDuration(Duration value) { leaseDuration = value; }
    public Duration getRequestTimeout() { return requestTimeout; }
    public void setRequestTimeout(Duration value) { requestTimeout = value; }
    public int getMaxAttempts() { return maxAttempts; }
    public void setMaxAttempts(int value) { maxAttempts = value; }
    public int getBatchSize() { return batchSize; }
    public void setBatchSize(int value) { batchSize = value; }
}

File: qqbot-module-qqbot-runtime/src/main/kotlin/com/mieai/qqbot/app/outbox/OutboxRuntimeProperties.kt
package com.mieai.qqbot.app.outbox

import java.time.Duration
import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties("qqbot.outbox")
class OutboxRuntimeProperties {
    var enabled: Boolean = true
    var pollInterval: Duration = Duration.ofSeconds(1)
    var leaseDuration: Duration = Duration.ofSeconds(45)
    var requestTimeout: Duration = Duration.ofSeconds(20)
    var maxAttempts: Int = 8
    var batchSize: Int = 16
}
